src/masternodes/rpc_proposals.cpp

```diff
@@ -0,0 +1,622 @@
+        if (IsValidDestination(dest)) {
+            coinControl.destChange = dest;
+        }
+    }
+
+    fund(rawTx, pwallet, optAuthTx, &coinControl);
+
+    // check execution
+    {
+        LOCK(cs_main);
+        CCoinsViewCache coins(&::ChainstateActive().CoinsTip());
+        if (optAuthTx)
+            AddCoins(coins, *optAuthTx, targetHeight);
+        auto metadata = ToByteVector(CDataStream{SER_NETWORK, PROTOCOL_VERSION, pm});
+        execTestTx(CTransaction(rawTx), targetHeight, metadata, CCreatePropMessage{}, coins);
+    }
+    return signsend(rawTx, pwallet, optAuthTx)->GetHash().GetHex();
+}
+
+UniValue vote(const JSONRPCRequest& request)
+{
+    CWallet* const pwallet = GetWallet(request);
+
+    RPCHelpMan{"vote",
+               "\nVote for community proposal" +
+               HelpRequiringPassphrase(pwallet) + "\n",
+               {
+                       {"proposalId", RPCArg::Type::STR, RPCArg::Optional::NO, "The proposal txid"},
+                       {"masternodeId", RPCArg::Type::STR, RPCArg::Optional::NO, "The masternode id which made the vote"},
+                       {"decision", RPCArg::Type::STR, RPCArg::Optional::NO, "The vote decision (yes/no/neutral)"},
+                       {"inputs", RPCArg::Type::ARR, RPCArg::Optional::OMITTED_NAMED_ARG, "A json array of json objects",
+                        {
+                                {"", RPCArg::Type::OBJ, RPCArg::Optional::OMITTED, "",
+                                 {
+                                         {"txid", RPCArg::Type::STR_HEX, RPCArg::Optional::NO, "The transaction id"},
+                                         {"vout", RPCArg::Type::NUM, RPCArg::Optional::NO, "The output number"},
+                                 },
+                                },
+                        },
+                       },
+               },
+               RPCResult{
+                       "\"hash\"                  (string) The hex-encoded hash of broadcasted transaction\n"
+               },
+               RPCExamples{
+                       HelpExampleCli("vote", "txid masternodeId yes")
+                       + HelpExampleRpc("vote", "txid masternodeId yes")
+               },
+    }.Check(request);
+
+    if (pwallet->chain().isInitialBlockDownload()) {
+        throw JSONRPCError(RPC_CLIENT_IN_INITIAL_DOWNLOAD,
+                           "Cannot vote while still in Initial Block Download");
+    }
+    pwallet->BlockUntilSyncedToCurrentChain();
+    LockedCoinsScopedGuard lcGuard(pwallet); // no need here, but for symmetry
+
+    RPCTypeCheck(request.params, { UniValue::VSTR, UniValue::VSTR, UniValue::VSTR, UniValue::VARR }, true);
+
+    auto propId = ParseHashV(request.params[0].get_str(), "proposalId");
+    auto mnId = ParseHashV(request.params[1].get_str(), "masternodeId");
+    auto vote = CPropVoteType::VoteNeutral;
+
+    auto voteStr = request.params[2].get_str();
+    if (voteStr == "no") {
```

Comment:
Can we check for uppercase "NO" instead to match CPropVoteToString return values ?
Right now, it takes 'no' as vote input and returns 'NO' in propVoteToJSON

Reply:
We can do it case insensitive, all other rpcs gets lowercase as well as when you type commands you don't need to add upper case, mostly.

Reply:
Ok I understand the logic. Would be nice to be case insensitive to have coherency between input and output